example/src/tests.ts

```diff
@@ -154,58 +159,279 @@ test('can message in a group', async () => {
+  // Create three MLS enabled Clients
+  const aliceClient = await Client.createRandom({
+    env: 'local',
+    enableAlphaMls: true,
+  })
+  const bobClient = await Client.createRandom({
+    env: 'local',
+    enableAlphaMls: true,
+  })
+  const camClient = await Client.createRandom({
+    env: 'local',
+    enableAlphaMls: true,
+  })
+
+  // Alice's num groups start at 0
+  let aliceGroups = await aliceClient.conversations.listGroups()
+  if (aliceGroups.length !== 0) {
+    throw new Error('num groups should be 0')
+  }
+
+  // Bob's num groups start at 0
+  let bobGroups = await bobClient.conversations.listGroups()
+  if (bobGroups.length !== 0) {
+    throw new Error('num groups should be 0')
+  }
+
+  // Cam's num groups start at 0
+  let camGroups = await camClient.conversations.listGroups()
+  if (camGroups.length !== 0) {
+    throw new Error('num groups should be 0')
+  }
+
+  // Alice creates a group
+  const aliceGroup = await aliceClient.conversations.newGroup([
+    bobClient.address,
+    camClient.address,
+  ])
+
+  // Alice's num groups == 1
+  aliceGroups = await aliceClient.conversations.listGroups()
+  if (aliceGroups.length !== 1) {
+    throw new Error('num groups should be 1')
+  }
+
+  // Alice can confirm memberAddresses
+  const memberAddresses = await aliceGroup.memberAddresses()
+  if (memberAddresses.length !== 3) {
+    throw new Error('num group members should be 3')
+  }
+  const lowercasedAddresses: string[] = memberAddresses.map((s) =>
+    s.toLowerCase()
+  )
+  if (
+    !(
+      lowercasedAddresses.includes(aliceClient.address.toLowerCase()) &&
+      lowercasedAddresses.includes(bobClient.address.toLowerCase())
+    )
+  ) {
+    throw new Error('missing address')
+  }
+
+  // Alice can send messages
+  await aliceGroup.send('hello, world')
+  await aliceGroup.send('gm')
+
+  // Bob's num groups == 1
+  await bobClient.conversations.syncGroups()
+  bobGroups = await bobClient.conversations.listGroups()
+  if (bobGroups.length !== 1) {
+    throw new Error(
+      'num groups for bob should be 1, but it is' + bobGroups.length
+    )
+  }
+
+  // Cam's num groups == 1
+  await camClient.conversations.syncGroups()
+  camGroups = await camClient.conversations.listGroups()
+  if (camGroups.length !== 1) {
+    throw new Error(
+      'num groups for cam should be 1, but it is' + camGroups.length
+    )
+  }
+
+  await aliceGroup.removeMembers([bobClient.address])
+  await aliceGroup.sync()
+  const aliceGroupMembers = await aliceGroup.memberAddresses()
+  if (aliceGroupMembers.length !== 2) {
+    throw new Error('num group members should be 2')
+  }
+
+  // await bobClient.conversations.syncGroups()
+
+  // bobGroups = await bobClient.conversations.listGroups()
+  // await bobGroups[0].sync()
+  // const bobGroupMessages = await bobGroups[0].messages()
+  // if (bobGroups.length !== 0) {
+  //   throw new Error(
+  //     'num groups for bob should be 0, but it is ' + bobGroups.length
+  //   )
+  // }
```

Comment:
I noticed that when this is uncommented that the test fails because the size of `bobClient.conversations.listGroups()` is still 1 after he is removed.

@richardhuaaa Is this expected that groups you have been removed from still are returned from listGroups()? If so, what is the proper way for someone to detect that they have been removed from a group?

Reply:
Hey @cameronvoell, yes with the current code the group and message history will remain in your database, but you should stop receiving updates. @neekolas is working on a way to notify the platform/app layer that a user has been removed from a group, but it is not ready yet.

> Is this expected that groups you have been removed from still are returned from listGroups()?

Alex was also confused about this, lmk if this seems unintuitive from the perspective of API design or if there's a better way to expose this. I don't think we've necessarily optimized around developer experience so far

Reply:
For now it's expected, a user will see past messages but not future

Reply:
@richardhuaaa I think from a UX perspective it's a bit confusing
From a developer perspective it makes sense if the purpose is to allow the developer freedom to clear the messages in some way.
Meaning if I as an integrating developer want to "remove" the user from the chat and clear messages I can, or if I want them to just stop receiving updates I have the options to do either

Reply:
> Hey cameronvoell, yes with the current code the group and message history will remain in your database, but you should stop receiving updates. neekolas is working on a way to notify the platform/app layer that a user has been removed from a group, but it is not ready yet.

> Alex was also confused about this, lmk if this seems unintuitive from the perspective of API design or if there's a better way to expose this. I don't think we've necessarily optimized around developer experience so far 

@richardhuaaa Sounds like it's just unintuitive because that one piece is not finished yet. Think the only real missing piece is a recommended way for someone to be notified that theyve been blocked, once we have that we should be able to handle the rest on the SDK side. Thanks for confirming!

Reply:
Agreed. I don't think it's our job to decide what to do _after_ the user has been removed. Some apps may want to continue showing the conversation and history, some won't. We just need to surface that information to developers. I'll work on adding that today.